runtime/autoload/provider/python3.vim

```diff
@@ -1,45 +1,16 @@
-" The Python3 provider uses a Python3 host to emulate an environment for running
-" python3 plugins. :help provider
-"
-" Associating the plugin with the Python3 host is the first step because
-" plugins will be passed as command-line arguments
-
 if exists('g:loaded_python3_provider')
   finish
 endif
-let [s:prog, s:err] = provider#pythonx#Detect(3)
-let g:loaded_python3_provider = empty(s:prog) ? 1 : 2
+let g:loaded_python3_provider = 1
 
-function! provider#python3#Prog() abort
-  return s:prog
+function! provider#python3#Call(method, args) abort
+  return v:lua.vim.provider.python.call(a:method, a:args)
```

Comment:
Can we use a normal `require('vim.provider.python')`? With this `vim.provider` is loaded on the first access, but right now it's going to load all (future) providers straight away, unless we also make those lazy loaded. And `require` should just be easier.
```suggestion
  return luaeval('require("vim.provider.python").call(_A[1], _A[2])', [a:method, a:args])
```

Reply:
If we are doing this, then we should make the `vim.provider.python` module lazy-loaded from `vim.provider` (which will be always executed on startup!).

Reply:
I'm already fine with removing `vim.provider` which is currently done, but another possible solution is to use lazy-loading in `vim.provider` as in `vim.treesitter`: See 
TreesitterModule, https://github.com/neovim/neovim/blob/master/runtime/lua/vim/treesitter.lua#L7-L13 so we don't need to do `require('vim.treesitter.python')` but simply `vim.treesitter`.